### PR TITLE
[New Rules] Docker Entrypoint Netcon / Nsenter Escape

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This PR adds two rules to extend coverage on Docker.

### Egress Connection from Entrypoint in Container
This rule identifies a sequence of events where a process named `entrypoint.sh` is started in a container, followed by a network connection attempt. This sequence indicates a potential egress connection from an entrypoint in a container. An entrypoint is a command or script specified in the Dockerfile and executed when the container starts. Attackers can use this technique to establish a foothold in the environment, escape from a container to the host, or establish persistence.

0 hits in telemetry last 90d, only TPs in my testing stack.

image

### Docker Escape via Nsenter
This rule identifies a UID change event via `nsenter`. The `nsenter` command is used to enter a namespace, which is a way to isolate processes and resources. Attackers can use `nsenter` to escape from a container to the host, which can lead to privilege escalation and lateral movement.

0 hits in telemetry last 7 days, only TPs (when removing cidr_match due to private IP testing):

image

### CODE REVIEW

File: rules/linux/execution_egress_connection_from_entrypoint_in_container.toml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    "Use Case: Threat Detection",
+    "Tactic: Execution",
+    "Data Source: Elastic Defend",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+sequence by host.id with maxspan=3s
+  [process where host.os.type == "linux" and event.type == "start" and event.action == "exec" and
+   process.entry_leader.entry_meta.type == "container" and process.name == "entrypoint.sh"] by process.entity_id
```
Comment (terrancedejesus) [inline rules/linux/execution_egress_connection_from_entrypoint_in_container.toml:35]: ohhhh very nice ⭐

File: rules/linux/execution_egress_connection_from_entrypoint_in_container.toml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Egress Connection from Entrypoint in Container"
+risk_score = 47
+rule_id = "c75d0c86-38d6-4821-98a1-465cff8ff4c8"
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
```
Comment (terrancedejesus) [inline rules/linux/execution_egress_connection_from_entrypoint_in_container.toml:24]: Should we add `Domain: Container` here as well? May help associate endpoint rules with docker unless free searching for "docker" itself.

File: rules/linux/privilege_escalation_docker_escape_via_nsenter.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+index = ["logs-endpoint.events.process*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Docker Escape via Nsenter"
+references = ["https://www.cyberark.com/resources/threat-research-blog/the-route-to-root-container-escape-using-kernel-exploitation"]
+risk_score = 47
+rule_id = "5f3ab3ce-7b41-4168-a06a-68d2af8ebc88"
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
```
Comment (terrancedejesus) [inline rules/linux/privilege_escalation_docker_escape_via_nsenter.toml:24]: Add `Domain: Container`?

**Review by terrancedejesus — APPROVED** [review]
Nice additions for Docker 🚀

**Review by imays11 — APPROVED** [review]

**Pull request merged**